### PR TITLE
Feature/#57 Highlight Set Bonus and Pieces

### PR DESCRIPTION
**Pull request by tealeon**
implements #57

### CODE REVIEW

**Comment by tealeon** [comment]
grafik

File: src/app/services/item-set.service.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+    // Now for the items that are not equipped, thus not highlighted
+    let difference = setBonus.pieces.filter(x => !tmp.includes(x));
+    for (let objectId of difference) {
+      const name = this.itemDataService.getItemDetail(objectId).name;
+      pieces.push({ name, isHighlighted: false });
+    }
+
+    result.pieces = [...new Map(pieces.map(item => [item['name'], item])).values()];
```
Comment (lorenzgruber) [inline src/app/services/item-set.service.ts:51]: I think this line can be simplified to `result.pieces = pieces;`

File: src/app/services/index.ts [hunk 2]
```diff
@@ -7,3 +7,5 @@ export * from './talent-data.service';
 export * from './skill-talent.service';
 export * from './selected-item.service';
 export * from './condition-data.service';
+export * from './extracted-data.service';
```
Comment (lorenzgruber) [inline src/app/services/index.ts:10]: Remove this line

**Review by lorenzgruber — CHANGES_REQUESTED** [review]

File: src/app/services/item-set.service.ts (hunk 1 above)
Comment (tealeon) [reply src/app/services/item-set.service.ts:51]: simplyfying this line leads to redundant display of required armor pieces for a set-bonus when e. g. two of the same rings are equipped

Comment (halilbahar) [reply src/app/services/item-set.service.ts:51]: Please provide a comment for this line with a link to this [Stackoverflow Answer](https://stackoverflow.com/a/58429784/11125147)

**Review by halilbahar — APPROVED** [review]

**Review by lorenzgruber — APPROVED** [review]

**Pull request merged**